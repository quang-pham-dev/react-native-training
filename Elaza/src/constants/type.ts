export const SIZE = {
  TINY: 'tiny',
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
  BASE: 'base',
}

export const HEADING_TYPE = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
}

export const PARAGRAPH_TYPE = {
  TINY: 'tiny',
  SMALL: 'small',
  BASE: 'base',
  MEDIUM: 'medium',
  LARGER: 'larger',
}

export const BUTTON_TYPE = {
  CONFIRM: 'confirm',
  CANCEL: 'cancel',
  LINK: 'link',
  SUBMIT: 'submit',
}
