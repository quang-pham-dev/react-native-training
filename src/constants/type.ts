export const SIZE = {
  TINY: 'tiny',
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
  BASE: 'base',
}

export const HEADING_TYPE = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
}

export const PARAGRAPH_TYPE = {
  TINY: 'tiny',
  SMALL: 'small',
  BASE: 'base',
  MEDIUM: 'medium',
  LARGER: 'larger',
  BRAND: 'brand',
  PRODUCT_NAME_AND_TYPE: 'productNameAndType',
  PRODUCT_PRICE: 'productPrice',
  ALREADY_ACCOUNT: 'alreadyAccount',
  ALREADY_SIGN_IN: 'alreadySignIn',
  CONFIRM_AGREE: 'confirmAgree',
  TERM_AND_CONDITION: 'termAndCondition',
  PROFILE_USERNAME: 'profileUsername',
  PROFILE_VERIFIED: 'profileVerified',
  PROFILE_ITEM: 'profileItem',
  BRAND_DETAIL_TITLE: 'brandDetailTitle',
  BRAND_DETAIL_ITEM_COUNT: 'brandDetailItemCount',
}

export const BUTTON_TYPE = {
  BOTTOM: 'bottom',
  SOCIAL: 'social',
}
