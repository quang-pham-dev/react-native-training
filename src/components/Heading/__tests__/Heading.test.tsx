import React from 'react'
import {create} from 'react-test-renderer'

import HeadingStyled from '../Heading.styles'

import {HEADING_TYPE} from '@constants/type'

describe('Heading', () => {
  test('should render Default type correctly', () => {
    const component = create(
      <HeadingStyled type={HEADING_TYPE.H1}>Sample h1</HeadingStyled>,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('should render Default type with custom props', () => {
    const component = create(
      <HeadingStyled type={HEADING_TYPE.H2}>Sample h2</HeadingStyled>,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
