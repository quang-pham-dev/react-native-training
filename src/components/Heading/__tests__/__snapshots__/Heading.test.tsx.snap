// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render Default type correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-Regular",
        "fontSize": 24,
        "lineHeight": 22,
        "textAlign": "center",
      },
    ]
  }
  type="h1"
>
  Sample h1
</Text>
`;

exports[`Heading should render Default type with custom props 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-Regular",
        "fontSize": 24,
        "lineHeight": 29,
        "textAlign": "center",
      },
    ]
  }
  type="h2"
>
  Sample h2
</Text>
`;
